compiler/resolution/implementForallIntents.cpp

```diff
@@ -1723,6 +1724,105 @@ void implementForallIntents2wrapper(CallExpr* call, CallExpr* eflopiHelper)
 // for the (new) ForallStmt-based representation.
 //
 
+// Forall Intent information about each outer variable.
+class FIinfo {
+public:
+  Symbol* ovar;       // outer variable
+  Symbol* svar;       // shadow variable
+  Symbol* reduceGVar; // reduction class, non-NULL only for reduce intents
+  Symbol* reduceMark; // transitional measure; TODO: remove
+  TFITag  intent;     // the forall intent for this variable
+
+  bool isReduce() { return intent == TFI_REDUCE; }
+
+  FIinfo(Symbol* outerVar):
+    ovar(outerVar), svar(NULL),
+    reduceGVar(NULL), reduceMark(NULL), intent(TFI_DEFAULT)
+  { }
+};
```

Comment:
At the moment this is a local helper data structure. Hopefully it will be obviated by IntentVariables. The way it is now, it will not help to make it into an AST node.

I will rename it to ForallIntentInfo.